packages/lib/server/repository/booking.ts

```diff
@@ -94,6 +95,9 @@ export class BookingRepository {
       attendees: { some: { noShow: false } },
       status: BookingStatus.ACCEPTED,
       eventTypeId,
+      createdAt: {
+        gte: dayjs().utc().subtract(28, "day").toDate(),
+      },
```

Comment:
The issue is described only for weighted round-robin booking, but doesn’t this filter out all round-robin bookings?

Reply:
`getAllBookingsForRoundRobin()` is used only for RR weight adjustment algorithm.

Reply:
It's also used for `leastRecentlyBookedUser()`, in that case we still want all bookings not just the last 28 days

Reply:
oh ok!, since the new paramerter 'bookingsOfAvailableUsers' for fn `leastRecntlyBookedUser()` was added from this PR https://github.com/calcom/cal.com/pull/15558  (RR Weights), I assumed even here 28days bookings is only required.
In that case will add one more parameter to fn `getAllBookingsForRoundRobin()` to accept input as like past number of days to fetch bookings and will default to fetch all bookings. And will rename to `getBookingsForRoundRobin()`.

Reply:
ok

Reply:
@CarinaWolli , there are 2 fn getBookingsForRoundRobin() calls from getLuckyUser.ts , for these two all bookings are fetched.
The other 2 remaining calls are from `addWeightAdjustmentToNewHosts()`, for these two only past 28 days are used.
Apart from these no calls to getBookingsForRoundRobin().
Hope this is fine.